weatherman.py

```diff
@@ -0,0 +1,194 @@
+import argparse
+import calendar
+from colorama import Fore, Style
+import csv
+import glob
+import os
+
+def convert_month(weather_month_input):
+    try:
+        month_number = int(weather_month_input)
+        if 1 <= month_number <= 12:
+            return calendar.month_abbr[month_number]
+    except ValueError:
+        raise ValueError("Invalid input")
+    
+def read_files(weather_year, weather_month):
+    weather_details = []
+
+    base_directory = r"D:\Cogent Labs\weatherfiles\weatherfiles"
+    pattern = f"*{weather_year}*"
+    if weather_month:
+        pattern = f"*{weather_year}*{weather_month}*"
+
+    files_to_process = glob.glob(os.path.join(base_directory, pattern))
+
+    for weathermas_file_path in files_to_process:
+        with open(weathermas_file_path, 'r') as weather_file:
+            reader = csv.DictReader(weather_file)
+            weather_details.extend(reader)
+    
+    return weather_details
+
+def calculate_statistics_year(weather_details):
```

Comment:
The same principle applies to functions as well; their names should accurately describe the actual functionality they perform internally.

`calculate_statistics_year` does not describe the actual functionality,

`calculate_yearly_weather_records`